db/document.go

```diff
@@ -402,14 +402,17 @@ func unmarshalDocument(docid string, data []byte) (*Document, error) {
 	return doc, nil
 }
 
-func unmarshalDocumentWithXattr(ctx context.Context, docid string, data []byte, xattrData []byte, userXattrData []byte, cas uint64, unmarshalLevel DocumentUnmarshalLevel) (doc *Document, err error) {
+func unmarshalDocumentWithXattr(ctx context.Context, docid string, data []byte, syncXattrData []byte, hlvXattrData []byte, userXattrData []byte, cas uint64, unmarshalLevel DocumentUnmarshalLevel) (doc *Document, err error) {
 
-	if xattrData == nil || len(xattrData) == 0 {
+	if (syncXattrData == nil || len(syncXattrData) == 0) && hlvXattrData == nil {
```

Comment:
This should be able to be simplified to

```suggestion
	if len(syncXattrData) == 0 && hlvXattrData == nil {
```

Reply:
It seems incorrect to be checking for zero length on one xattr and nil on the other?

Reply:
I'd probably just change this to len checks for both in case someone passes a literal `[]byte{}` in an argument and len(nil) is valid.